tox_pdm/plugin4.py

```diff
@@ -0,0 +1,232 @@
+
+
+@impl
+def tox_register_tox_env(register: ToxEnvRegister) -> t.Optional[bool]:
+    register.add_run_env(PdmRunner)
+    register.add_package_env(PdmPackageEnv)
+    register.default_run_env = "pdm"
+
+
+class Pdm(Python):
+    def __init__(self, create_args: ToxEnvCreateArgs) -> None:
+        self._executor: t.Optional[Execute] = None
+        self._installer: t.Optional[Pip] = None
+        self._package_path: t.Optional[Path] = None
+        super().__init__(create_args)
+
+    @property
+    def executor(self) -> Execute:
+        if not self._executor:
+            self._executor = PDMRunExecutor(self.options.is_colored)
+        return self._executor
+
+    @property
+    def installer(self) -> Pip:
+        if self._installer is None:
+            self._installer = PipPep582(self)
+        return self._installer
+
+    @property
+    def package_path(self) -> Path:
+        if not self._package_path:
+            self._package_path = Path(
+                get_env_lib_path(self.options.pdm, self.env_dir)
+            ).parent
+        return self._package_path
+
+    @property
+    def runs_on_platform(self) -> str:
+        return sys.platform
+
+    def create_python_env(self) -> None:
+        clone_pdm_files(self.env_dir, self.core["toxinidir"])
+        self.execute(
+            ["pdm", "use", "-f", self.base_python.extra["executable"]], StdinSource.OFF
+        )
+
+    def _get_python(self, base_python: t.List[str]) -> t.Optional[PythonInfo]:
+        interpreter = next(
+            filter(None, (get_interpreter(p, []) for p in base_python)), None
+        )
+        if not interpreter:
+            return None
+        return PythonInfo(
+            implementation=interpreter.implementation,
+            version_info=interpreter.version_info,
+            version=interpreter.version,
+            is_64=(interpreter.architecture == 64),
+            platform=interpreter.platform,
+            extra={"executable": Path(interpreter.system_executable)},
+        )
+
+    def python_cache(self) -> t.Dict[str, t.Any]:
+        base = super().python_cache()
+        base.update({"executable": str(self.base_python.extra["executable"])})
+        return base
+
+    def env_bin_dir(self) -> Path:
+        return self.package_path / ("Scripts" if os.name == "nt" else "bin")
+
+    def env_python(self) -> Path:
+        return t.cast(Path, self.base_python.extra["executable"])
+
+    def env_site_package_dir(self) -> Path:
+        return self.package_path / "lib"
+
+    def prepend_env_var_path(self) -> t.List[Path]:
+        return [self.env_bin_dir]
+
+
+class PdmRunner(Pdm, PythonRun):
+    def _setup_env(self) -> None:
+        super()._setup_env()
+        groups = self.conf["groups"]
+        cmd = ["pdm", "install", "--no-self"]
+        for group in groups:
+            cmd.extend(("--group", group))
+        self.execute(cmd, StdinSource.OFF)
+        bin_path = self.env_site_package_dir() / "bin"
```

Comment:
Always bin? Not scripts on Windows?

Reply:
It is the pattern of `pip install --target`

Reply:
Can you expand on this?

Reply:
If you do `pip install --target mypath django`, the scripts will be created under `mypath/bin`, on Windows and POSIX systems.

Reply:
Did not know, thanks for clarifying. Perhaps would be valuable to add this as a comment to the code.